### ISSUE TITLE
Yield an error object instead of exception when possible

### ISSUE THREAD

**Issue by halaxa** [halaxa/json-machine#8]
The idea is not to throw exception if a parse error occurs inside of the scructure which is about to be yielded. Instead, some kind of parse error object could be yielded so consumer can decide whether to stop iteration or just to skip the errorneous structure and continue.

Usage:
```php
foreach(JsonMachine::fromFile('a.json') as $key => $value) {
    if ($key instanceof JsonError || $value instanceof JsonError) {
        // continue / log / throw ...
    }
    // process $key, $value
}
```

**Comment by halaxa** [comment]
It must wait to version 1.0 due to upcoming changes in `Parser`

**Comment by halaxa** [comment]
Implemented on master. See `ErrorWrappingDecoder` in README.md. Can anyone try it? @DevinNorgarb?

**Comment by halaxa** [comment]
Implemented in `0.5.0`
